src/Expensify.js

```diff
@@ -114,118 +144,77 @@ class Expensify extends PureComponent {
         migrateOnyx()
             .then(() => {
                 // In case of a crash that led to disconnection, we want to remove all the push notifications.
-                if (!this.isAuthenticated()) {
+                if (!isAuthenticated()) {
                     PushNotification.clearNotifications();
                 }
 
-                this.setState({isOnyxMigrated: true});
+                setIsOnyxMigrated(true);
             });
 
-        this.appStateChangeListener = AppState.addEventListener('change', this.initializeClient);
+        appStateChangeListener.current = AppState.addEventListener('change', initializeClient);
 
         // Open chat report from a deep link (only mobile native)
         Linking.addEventListener('url', state => Report.openReportFromDeepLink(state.url));
-    }
 
-    componentDidUpdate() {
-        if (!this.state.isNavigationReady || !this.state.isSplashShown) {
+        return () => {
+            if (!appStateChangeListener.current) { return; }
+            appStateChangeListener.current.remove();
+        };
+    }, []);
+
+    React.useEffect(() => {
+        if (!isNavigationReady || !isSplashShown) {
             return;
         }
 
-        const shouldHideSplash = !this.isAuthenticated() || this.props.isSidebarLoaded;
+        const shouldHideSplash = !isAuthenticated() || props.isSidebarLoaded;
 
         if (shouldHideSplash) {
             BootSplash.hide();
 
-            // eslint-disable-next-line react/no-did-update-set-state
-            this.setState({isSplashShown: false});
+            setIsSplashShown(false);
 
             // If the app is opened from a deep link, get the reportID (if exists) from the deep link and navigate to the chat report
             Linking.getInitialURL().then(url => Report.openReportFromDeepLink(url));
         }
-    }
+    });
```

Comment:
Shouldn't we add `isNavigationReady` and  `isSplashShown` to the dependencies array here? We're also calling `isAuthenticated` and I think that should also be a dependency, but since it's a function I'm not sure how it can be done. Also, if we keep the dependency array empty, this will only run once, so this will essentially be working like `componentDidMount`.

Reply:
On second thought I think we're good about `isAuthenticated`, let me know your thoughts!

Reply:
Good catch! Those definitely should be added to ensure that this isn't called every render. I've updated them.

I agree that we're good with `isAuthenticated` because that result comes from calling a function.